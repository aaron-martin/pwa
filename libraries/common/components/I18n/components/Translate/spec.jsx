/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import I18n from '../../index';

describe('<Translate />', () => {
  let renderedElement;
  const testLocales = {
    greeting: 'Hello {name}',
  };

  /**
   * Renders the component.
   * @param {Object} props The component props.
   */
  const renderComponent = (props) => {
    renderedElement = mount((
      <I18n.Provider {...props}>
        <I18n.Text string="greeting" params={{ name: 'Test' }} />
      </I18n.Provider>
    ));
  };

  beforeEach(() => {
    renderComponent({
      lang: 'en-US',
      locales: testLocales,
    });
  });

  describe('Given the component was mounted to the DOM', () => {
    it('should match snapshot', () => {
      expect(renderedElement).toMatchSnapshot();
    });

    it('should render translated text', () => {
      const text = renderedElement.find(I18n.Text).text();
      expect(text).toBe('Hello Test');
    });
  });
});
