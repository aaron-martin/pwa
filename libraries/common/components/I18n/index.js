/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import I18nProvider from './components/I18nProvider';
import Placeholder from './components/Placeholder';
import Translate from './components/Translate';
import FormatDate from './components/FormatDate';
import FormatTime from './components/FormatTime';
import FormatPrice from './components/FormatPrice';

export default {
  Provider: I18nProvider,
  Text: Translate,
  Date: FormatDate,
  Time: FormatTime,
  Placeholder,
  Price: FormatPrice,
};
