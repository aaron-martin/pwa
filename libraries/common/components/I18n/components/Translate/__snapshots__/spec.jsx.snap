// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Translate /> Given the component was mounted to the DOM should match snapshot 1`] = `
<I18nProvider
  lang="en-US"
  locales={
    Object {
      "greeting": "Hello {name}",
    }
  }
>
  <Translate
    className=""
    params={
      Object {
        "name": "Test",
      }
    }
    string="greeting"
  >
    <span
      className=""
    >
      Hello Test
    </span>
  </Translate>
</I18nProvider>
`;
